control-plane/pkg/reconciler/trigger/v2/triggerv2.go

```diff
@@ -138,13 +138,15 @@ func (r ReconcilerV2) reconcileConsumerGroup(ctx context.Context, trigger *event
 		},
 	}
 
-	//todo validate newcg
-
 	cg, err := r.ConsumerGroupLister.ConsumerGroups(trigger.GetNamespace()).Get(string(trigger.UID)) //Get by consumer group name
 	if err != nil && !apierrors.IsNotFound(err) {
 		return nil, err
 	}
 
+	if err := newcg.Validate(ctx); err != nil {
+		return nil, fmt.Errorf("failed to validate expected consumer group %s/%s: %w", newcg.GetNamespace(), newcg.GetName(), err)
+	}
```

Comment:
I think there shouldn't validate the CG here since if it's invalid we get an error once we create it.

Reply:
The idea here was to validate the `newcg` expected consumer group respresenation that we build before calling `.Create(ctx....)`. Such that `newcg` must have all the fields expected by Validate to be set. Shouldn't that be necessary?

Reply:
Right now, we aren't setting `Replicas`, `Selector`, etc in `newcg` so Validate will fail..

Reply:
> Shouldn't that be necessary?

Yes, since the webhook will reject invalid consumer groups when we call `Create(ctx, ...)`.

The other problem with this is that a resource is validated after calling `SetDefaults(ctx)` on it which is done by the webhook as well, so it's ok to call `Create(ctx, ...)` with an invalid resource since after the mutating webhook runs the resource might become valid.

Reply:
I see, now I understand how the SetDefaults and Validate apis are used. I'm following. Thanks @pierDipi.